Classes/YTPlayerView.m

```diff
@@ -104,314 +104,355 @@ - (BOOL)loadWithPlaylistId:(NSString *)playlistId playerVars:(NSDictionary *)pla
+  NSString *command = [NSString stringWithFormat:@"player.loadVideoByUrl('%@', %@, %@);",
+      videoURL, startSecondsValue, endSecondsValue];
+  [self evaluateJavaScript:command];
 }
 
 #pragma mark - Cueing methods for lists
 
 - (void)cuePlaylistByPlaylistId:(NSString *)playlistId
                           index:(int)index
-                   startSeconds:(float)startSeconds
-               suggestedQuality:(YTPlaybackQuality)suggestedQuality {
+                   startSeconds:(float)startSeconds {
   NSString *playlistIdString = [NSString stringWithFormat:@"'%@'", playlistId];
   [self cuePlaylist:playlistIdString
                  index:index
-          startSeconds:startSeconds
-      suggestedQuality:suggestedQuality];
+          startSeconds:startSeconds];
 }
 
 - (void)cuePlaylistByVideos:(NSArray *)videoIds
                       index:(int)index
-               startSeconds:(float)startSeconds
-           suggestedQuality:(YTPlaybackQuality)suggestedQuality {
+               startSeconds:(float)startSeconds {
   [self cuePlaylist:[self stringFromVideoIdArray:videoIds]
                  index:index
-          startSeconds:startSeconds
-      suggestedQuality:suggestedQuality];
+          startSeconds:startSeconds];
 }
 
 - (void)loadPlaylistByPlaylistId:(NSString *)playlistId
                            index:(int)index
-                    startSeconds:(float)startSeconds
-                suggestedQuality:(YTPlaybackQuality)suggestedQuality {
+                    startSeconds:(float)startSeconds {
   NSString *playlistIdString = [NSString stringWithFormat:@"'%@'", playlistId];
   [self loadPlaylist:playlistIdString
                  index:index
-          startSeconds:startSeconds
-      suggestedQuality:suggestedQuality];
+          startSeconds:startSeconds];
 }
 
 - (void)loadPlaylistByVideos:(NSArray *)videoIds
                        index:(int)index
-                startSeconds:(float)startSeconds
-            suggestedQuality:(YTPlaybackQuality)suggestedQuality {
+                startSeconds:(float)startSeconds {
   [self loadPlaylist:[self stringFromVideoIdArray:videoIds]
                  index:index
-          startSeconds:startSeconds
-      suggestedQuality:suggestedQuality];
+          startSeconds:startSeconds];
 }
 
 #pragma mark - Setting the playback rate
 
-- (float)playbackRate {
-  NSString *returnValue = [self stringFromEvaluatingJavaScript:@"player.getPlaybackRate();"];
-  return [returnValue floatValue];
+- (void)playbackRate:(_Nullable YTFloatCompletionHandler)completionHandler {
+  [self evaluateJavaScript:@"player.getPlaybackRate();"
+         completionHandler:^(id  _Nullable result, NSError * _Nullable error) {
+    if (!completionHandler) {
+      return;
+    }
+    if (error) {
+      completionHandler(-1, error);
+      return;
+    }
+    if (!result || ![result isKindOfClass:[NSNumber class]]) {
+      completionHandler(0, nil);
+    }
+    completionHandler([result floatValue], nil);
```

Comment:
In case if the `result` is not an `NSNumber` instance, this may crash (in case of result does not have `floatValue` function) or `completionHandler` will be called twice(otherwise).
either need to return in the `if` or need an `else`?

Reply:
thanks for noticing, will be posting a fix for this soon.